src/Data/Inputs.elm

```diff
@@ -50,11 +50,24 @@ type alias CustomCountryMixes =
 
 fromQuery : Db -> Query -> Result String Inputs
 fromQuery db query =
+    let
+        material =
+            db.materials |> Material.findByUuid query.material
+    in
     Ok Inputs
         |> RE.andMap (Ok query.mass)
-        |> RE.andMap (db.materials |> Material.findByUuid query.material)
+        |> RE.andMap material
         |> RE.andMap (db.products |> Product.findById query.product)
-        |> RE.andMap (db.countries |> Country.findByCodes query.countries)
+        |> RE.andMap
+            (material
+                |> Result.andThen
+                    (\{ defaultCountry } ->
+                        query.countries
+                            -- Update the list of countries: the first country (from the material step) is constrained to be the material's default country
+                            |> updateCountryList 0 defaultCountry
+                            |> (\updatedCountries -> Country.findByCodes updatedCountries db.countries)
+                    )
+            )
```

Comment:
Two things:
- I wonder if this shouldn't/couldn't be extracted to a dedicated function?
- We need to get rid of the duplicate operation performed in Page.Simulator.update when material is changed from the the Web UI: https://github.com/MTES-MCT/wikicarbone/blob/master/src/Page/Simulator.elm#L301.

Reply:
Yes I extracted the function.
Regarding the duplicate operation, what exactly are you talking about?

The code in `Simulator.elm` is in charge of resetting the recycled ratio from what I can see

Reply:
Ahh yes my bad, scratch that